tests/integration/credset_test.go

```diff
@@ -0,0 +1,516 @@
+				cs.ObjectMeta.Namespace = ns
+				cred := porterv1.Credential{
+					Name: "insecureValue",
+					Source: porterv1.CredentialSource{
+						Secret: name,
+					},
+				}
+				cs.Spec.Credentials = append(cs.Spec.Credentials, cred)
+				cs.Spec.Namespace = ns
+
+				Expect(k8sClient.Create(ctx, cs)).Should(Succeed())
+				Expect(waitForPorterCS(ctx, cs, "waiting for credential set to apply")).Should(Succeed())
+				validateCredSetConditions(cs)
+
+				Log("install porter-test-me bundle with new credset")
+				inst := NewTestInstallation("cs-with-secret")
+				inst.ObjectMeta.Namespace = ns
+				inst.Spec.Namespace = ns
+				inst.Spec.CredentialSets = append(inst.Spec.CredentialSets, name)
+				inst.Spec.SchemaVersion = "1.0.0"
+				Expect(k8sClient.Create(ctx, inst)).Should(Succeed())
+				Expect(waitForPorter(ctx, inst, "waiting for porter-test-me to install")).Should(Succeed())
+				validateInstallationConditions(inst)
+
+				// Validate that the correct credential set was used by the installation
+				jsonOut := runAgentAction(ctx, "show-outputs", ns, []string{"installation", "outputs", "list", "-n", ns, "-i", inst.Spec.Name, "-o", "json"})
+				credsValue := gjson.Get(jsonOut, `#(name=="outInsecureValue").value`).String()
+				Expect(credsValue).To(Equal(testSecret))
+			})
+		})
+	})
+})
+var _ = Describe("CredentialSet secret does not exist", func() {
+	Context("when an installation is created with a CredentialSet resource that references a secret that does not exist", func() {
+		It("should fail the installation install", func() {
+			ctx := context.Background()
+			ns := createTestNamespace(ctx)
+			name := "test-cs-" + ns
+			By("successfully creating the credential set", func() {
+				Log(fmt.Sprintf("create credential set '%s' for credset", name))
+				cs := NewTestCredSet(name)
+				cs.ObjectMeta.Namespace = ns
+				cred := porterv1.Credential{
+					Name: "insecureValue",
+					Source: porterv1.CredentialSource{
+						Secret: name,
+					},
+				}
+				cs.Spec.Credentials = append(cs.Spec.Credentials, cred)
+				cs.Spec.Namespace = ns
+
+				Expect(k8sClient.Create(ctx, cs)).Should(Succeed())
+				Expect(waitForPorterCS(ctx, cs, "waiting for credential set to apply")).Should(Succeed())
+				validateCredSetConditions(cs)
+
+			})
+			By("failing the installation install", func() {
+				Log("install porter-test-me bundle with new credset")
+				inst := NewTestInstallation("cs-no-secret")
+				inst.ObjectMeta.Namespace = ns
+				inst.Spec.Namespace = ns
+				inst.Spec.CredentialSets = append(inst.Spec.CredentialSets, name)
+				inst.Spec.SchemaVersion = "1.0.0"
+				Expect(k8sClient.Create(ctx, inst)).Should(Succeed())
+				err := waitForPorter(ctx, inst, "waiting for porter-test-me to install")
+				Expect(err).Should(HaveOccurred())
+				validateInstallationConditions(inst)
+				Expect(inst.Status.Phase).To(Equal(porterv1.PhaseFailed))
+			})
+		})
+	})
+})
+
+// TODO: Add failed installation due to missing secret test
```

Comment:
The todo here is referencing functionality in porter, not the operator, since as we discussed at the last meeting, the operator isn't aware of or validating what exists before calling porter.

Maybe switch this todo to an issue and describe the functionality you'd like to see in the operator here.